homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -29,35 +35,67 @@ class MotionBlindsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
 
     def __init__(self):
         """Initialize the Motion Blinds flow."""
-        self.host = None
-        self.key = None
+        self._host = None
+        self._key = None
+        self._ips = []
 
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            self.host = user_input[CONF_HOST]
-            self.key = user_input[CONF_API_KEY]
-            return await self.async_step_connect()
+            self._host = user_input.get(CONF_HOST)
+
+            if self._host is not None:
+                return await self.async_step_connect()
+
+            # Use MotionGateway discovery
+            discover_class = MotionDiscovery()
+            gateways = await self.hass.async_add_executor_job(discover_class.discover)
+            self._ips = list(gateways)
```

Comment:
Is MotionDiscovery.discover() guaranteed to return an iterable if it fails or doesn't discover anything?  My instinct would be try/except or check for None.

Reply:
Also wonder if discover has a default timeout?

Reply:
I had a similar thought and poked into the library here:
https://github.com/starkillerOG/motion-blinds/blob/e5ae3ca0b55213ba4b36d9940650fa31d8f94f51/motionblinds/motion_blinds.py#L124
Given this is just doing a multicast send and receive it looked ok to me to just catch the timeout, but I didn't look super close. I think it looks good on both points.

Reply:
Yes `MotionDiscovery.discover()` is garanteed to return `{}` or a dict of gateways.
Inside the discovery there is a large `try: except:` that catches any error and stops discovery and returns the then discovered gateways or `{}`.

`MotionDiscovery.discover()` has a default timeout of 10.0 seconds which can optionally be changed by using `MotionDiscovery.discover(discovery_time=25.0)`

Reply:
10 seconds sounds good to me - is there a HA standard?

Reply:
by the way, it always waits for the full timeout to occur before finishing because multiple gateways can be discovered (if one gateway is found, it is not garantueed that another gateway will not respond later).

So the `MotionDiscovery.discover(discovery_time=25.0)` will always take `discovery_time` to finish

Reply:
Not sure if there is a HA standard, but I expect not since it depends on the integration/devices how fast they are expected to respond on such a thing. So a diffrent timeout may be logical for diffrent types of devices...